### ISSUE TITLE
NameError - path method not defined

### ISSUE THREAD

**Issue by mike-stewart** [rafaelsales/whenever-test#2]
My schedule.rb file makes use of the Whenever `path` method, which either uses a user-specified value or the default defined here: https://github.com/javan/whenever/blob/efc759d10013e3df2ceb2a86c42a0292c55d4358/lib/whenever.rb#L19

However, it appears that when parsing this schedule file with Whenever::Test::Schedule, the method is not defined.

```
NameError: undefined local variable or method `path' for #<Whenever::Test::DSLInterpreter:0x007ff1ce8bebc0>
~/.rvm/gems/ruby-2.1.7@introhive/gems/whenever-test-1.0.0/lib/whenever/test/schedule.rb:23:in `instance_eval'
~/.rvm/gems/ruby-2.1.7@introhive/gems/whenever-test-1.0.0/lib/whenever/test/schedule.rb:23:in `instance_eval'
~/.rvm/gems/ruby-2.1.7@introhive/gems/whenever-test-1.0.0/lib/whenever/test/schedule.rb:23:in `parse'
~/.rvm/gems/ruby-2.1.7@introhive/gems/whenever-test-1.0.0/lib/whenever/test/schedule.rb:12:in `initialize'
```

**Comment by rafaelsales** [comment]
@mike-stewart IIRC, the `path` is set via the `set` dsl command. Example:

```ruby
      set :job_template, nil
      set :environment, :silly
      set :path, '/my/path'

      every 2.hours do
        runner "blahblah"
      end
```

Are your path being set like that? If not, could you provide me a bit more context on what you have in your schedule file?

**Comment by mike-stewart** [comment]
Mine isn't setting the `path` value via the DSL.
 
I.e. the schedule.rb has this line
```
set :output, "#{path}/log/cron_tasks.log"
```

This works, as whenever has a default value. As per the whenever README:

> If a :path is not set it will default to the directory in which whenever was executed.

**Comment by rafaelsales** [comment]
@mike-stewart oh right, I missed that scenario.

That should be a very simple fix, probably as simple as adding the following method to https://github.com/rafaelsales/whenever-test/blob/master/lib/whenever/test/dsl.rb

```ruby
def path
  @path || ''
end
```

Feel free to submit a PR, making sure to add or adapt one of the specs to catch this.
I'm gonna do it tonight if you can't.
Thanks for reporting.

**Comment by mike-stewart** [comment]
Sure, I'll submit a PR this weekend.

**Comment by rafaelsales** [comment]
Thanks, looking forward :)

**Comment by mike-stewart** [comment]
After digging in, it turns out that the path variable I was using was being set in whenever's setup.rb file: https://github.com/javan/whenever/blob/master/lib/whenever/setup.rb

The problem was:
 1. Variables set in setup.rb were not available when parsing the schedule file, because only user-specified variables were set as instance variables in the DSL.
 1. There were no `attr_reader`s, so even if the `@path` variable was set explicitly, it was not possible to call `path`.
 1. The setup.rb was being parsed after setting user-specified variables, so it was necessary to change this behaviour to parse the setup.rb first, to avoid breaking the interface (and one of the tests).

I've proposed a fix in #3 - interested in seeing what you think.
